src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -255,7 +303,47 @@ private async Task AnalyzeProject(Solution solution, IGrouping<ProjectId, Docume
             }
         }
 
-        private async Task<ImmutableArray<Diagnostic>> AnalyzeDocument(Project project, ImmutableArray<DiagnosticAnalyzer> allAnalyzers, Compilation compilation, AnalyzerOptions workspaceAnalyzerOptions, Document document)
+        private async Task<ImmutableArray<Diagnostic>> AnalyzeProjectAsync(Project project, ImmutableArray<DiagnosticAnalyzer> analyzers, Compilation compilation, AnalyzerOptions workspaceAnalyzerOptions)
+        {
+            try
+            {
+                // There's real possibility that bug in analyzer causes analysis hang at document.
+                var perDocumentTimeout =
+                    new CancellationTokenSource(_options.RoslynExtensionsOptions.DocumentAnalysisTimeoutMs);
+
+                var diagnostics = ImmutableArray<Diagnostic>.Empty;
+
+                // Only basic syntax check is available if file is miscellanous like orphan .cs file.
+                // Those projects are on hard coded virtual project
+                if (project.Name != $"{Configuration.OmniSharpMiscProjectName}.csproj" && analyzers.Any())
+                {
+                    var compilationWithAnalyzers = compilation.WithAnalyzers(analyzers, new CompilationWithAnalyzersOptions(
+                        workspaceAnalyzerOptions,
+                        onAnalyzerException: OnAnalyzerException,
+                        concurrentAnalysis: true,
```

Comment:
we always had it set to false. I am a bit worried that setting it to true will cause even more people to complain about the Mono process CPU usage. do you see any performance improvements when this is set to true?

Reply:
Let me dig into this. My assumption is that running them serially would prolong the load on the CPU, but I don't want to regress.

Reply:
Going to revert this for now. Would like to look again at performance some time soon.